### ISSUE TITLE
Broken sk_region_intersect

### ISSUE THREAD

**Issue by RobotoRaccoon** [jb-aero/SKCompat#2]
[13:01:55 INFO]: An unexpected exception occured during the execution of a script.
[13:01:55 WARN]: java.lang.NoSuchMethodError: com.sk89q.worldguard.protection.regions.ProtectedRegion.getIntersectingRegions(Ljava/util/List;)Ljava/util/
List;
[13:01:55 WARN]:        at com.zeoldcraft.skcompat.SKWorldGuard$sk_region_intersect.exec(SKWorldGuard.java:390)
[13:01:55 WARN]:        at com.laytonsmith.core.Script.eval(Script.java:353)
[13:01:55 WARN]:        at com.laytonsmith.core.Script.eval(Script.java:322)
[13:01:55 WARN]:        at com.laytonsmith.core.Script.eval(Script.java:322)
[13:01:55 WARN]:        at com.laytonsmith.core.MethodScriptCompiler.execute(MethodScriptCompiler.java:1908)
[13:01:55 WARN]:        at com.laytonsmith.core.MethodScriptCompiler.execute(MethodScriptCompiler.java:1865)
[13:01:55 WARN]:        at com.laytonsmith.core.Script.run(Script.java:194)
[13:01:55 WARN]:        at com.laytonsmith.core.AliasCore.alias(AliasCore.java:169)
[13:01:55 WARN]:        at com.laytonsmith.core.functions.Meta$call_alias.exec(Meta.java:504)
[13:01:55 WARN]:        at com.laytonsmith.core.Script.eval(Script.java:353)
[13:01:55 WARN]:        at com.laytonsmith.core.MethodScriptCompiler.execute(MethodScriptCompiler.java:1908)
[13:01:55 WARN]:        at com.laytonsmith.core.MethodScriptCompiler.execute(MethodScriptCompiler.java:1865)
[13:01:55 WARN]:        at com.laytonsmith.core.Script.run(Script.java:194)
[13:01:55 WARN]:        at com.laytonsmith.core.AliasCore.alias(AliasCore.java:169)
[13:01:55 WARN]:        at com.laytonsmith.commandhelper.CommandHelperListener.runAlias(CommandHelperListener.java:91)
[13:01:55 WARN]:        at com.laytonsmith.commandhelper.CommandHelperListener.onPlayerCommandPreprocess(CommandHelperListener.java:140)
[13:01:55 WARN]:        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
[13:01:55 WARN]:        at sun.reflect.NativeMethodAccessorImpl.invoke(Unknown Source)
[13:01:55 WARN]:        at sun.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)
[13:01:55 WARN]:        at java.lang.reflect.Method.invoke(Unknown Source)
[13:01:55 WARN]:        at org.bukkit.plugin.java.JavaPluginLoader$1.execute(JavaPluginLoader.java:301)
[13:01:55 WARN]:        at org.bukkit.plugin.RegisteredListener.callEvent(RegisteredListener.java:62)
[13:01:55 WARN]:        at org.bukkit.plugin.SimplePluginManager.fireEvent(SimplePluginManager.java:502)
[13:01:55 WARN]:        at org.bukkit.plugin.SimplePluginManager.callEvent(SimplePluginManager.java:487)
[13:01:55 WARN]:        at net.minecraft.server.v1_8_R1.PlayerConnection.handleCommand(PlayerConnection.java:1107)
[13:01:55 WARN]:        at net.minecraft.server.v1_8_R1.PlayerConnection.a(PlayerConnection.java:950)
[13:01:55 WARN]:        at net.minecraft.server.v1_8_R1.PacketPlayInChat.a(PacketPlayInChat.java:26)
[13:01:55 WARN]:        at net.minecraft.server.v1_8_R1.PacketPlayInChat.a(PacketPlayInChat.java:53)
[13:01:55 WARN]:        at net.minecraft.server.v1_8_R1.PacketHandleTask.run(SourceFile:13)
[13:01:55 WARN]:        at java.util.concurrent.Executors$RunnableAdapter.call(Unknown Source)
[13:01:55 WARN]:        at java.util.concurrent.FutureTask.run(Unknown Source)
[13:01:55 WARN]:        at net.minecraft.server.v1_8_R1.MinecraftServer.z(MinecraftServer.java:683)
[13:01:55 WARN]:        at net.minecraft.server.v1_8_R1.DedicatedServer.z(DedicatedServer.java:316)
[13:01:55 WARN]:        at net.minecraft.server.v1_8_R1.MinecraftServer.y(MinecraftServer.java:623)
[13:01:55 WARN]:        at net.minecraft.server.v1_8_R1.MinecraftServer.run(MinecraftServer.java:526)
[13:01:55 WARN]:        at java.lang.Thread.run(Unknown Source)

**Comment by jb-aero** [comment]
Hmm what version of CH/WE/WG? If this is the case it shouldn't have compiled.

**Comment by RobotoRaccoon** [comment]
[09:33:56 INFO]: [WorldEdit] Loading WorldEdit v6.0;3342-78f975b9
[09:33:57 INFO]: [WorldGuard] Loading WorldGuard v6.0.0-SNAPSHOT.1512-
[09:33:57 INFO]: [CommandHelper] Loading CommandHelper v3.3.1-SNAPSHOT.2817-

**Comment by Pieter12345** [comment]
I've also found this bug. I've scripted this code to perform the same, feel free to use it:

# Function to return wether 2 regions intersect or not.

# The function assumes both coordSets are in the same world.

proc(_regions_intersect, @coordSet1, @coordSet2,

```
# Get the required variables
@max_x1 = max(@coordSet1[0][0], @coordSet1[1][0])
@max_y1 = max(@coordSet1[0][1], @coordSet1[1][1])
@max_z1 = max(@coordSet1[0][2], @coordSet1[1][2])

@min_x1 = min(@coordSet1[0][0], @coordSet1[1][0])
@min_y1 = min(@coordSet1[0][1], @coordSet1[1][1])
@min_z1 = min(@coordSet1[0][2], @coordSet1[1][2])

@max_x2 = max(@coordSet2[0][0], @coordSet2[1][0])
@max_y2 = max(@coordSet2[0][1], @coordSet2[1][1])
@max_z2 = max(@coordSet2[0][2], @coordSet2[1][2])

@min_x2 = min(@coordSet2[0][0], @coordSet2[1][0])
@min_y2 = min(@coordSet2[0][1], @coordSet2[1][1])
@min_z2 = min(@coordSet2[0][2], @coordSet2[1][2])

# Return wether the 2 regions intersect or not (It doesn't if one of the axis has no possible intersection already)
return(
    !(
        (@max_x1 < @min_x2 || @max_x2 < @min_x1) || # Checks x intersect
        (@max_y1 < @min_y2 || @max_y2 < @min_y1) || # Checks y intersect
        (@max_z1 < @min_z2 || @max_z2 < @min_z1)    # Checks z intersect
    )
)
```

)

# Function that returns an array of regions which overlap with @region in @world.

proc(_sk_region_intersect_custom, @world, @region,
    @regionCoordSet = sk_region_info(@region, @world)[0]
    @intersectionList = array()
    foreach(sk_all_regions(@world), @otherRegion,
        if(_regions_intersect(@regionCoordSet, sk_region_info(@otherRegion, @world)[0]),
            array_push(@intersectionList, @otherRegion)
        )
    )
    return @intersectionList
)

**Comment by PseudoKnight** [comment]
Correction: This was fixed in 1.0.2.
